pkg/controller/kogitoapp/resource/requested_resources.go

```diff
@@ -62,30 +62,43 @@ func GetRequestedResources(kogitoApp *v1alpha1.KogitoApp, client *client.Client)
 }
 
 func buildConfigS2IBuilder(chain *builderChain) *builderChain {
+	// if there is gitURI create a s2i build config.
 	if !chain.KogitoApp.Spec.IsGitURIEmpty() {
 		bc, err := newBuildConfigS2I(chain.KogitoApp)
 		if err != nil {
 			chain.Error = err
 			return chain
 		}
+		chain.Resources.BuildConfigS2I = &bc
 
+		// otherwise create a s2i build from file
+	} else {
+		bc, err := newBuildConfigS2IFromFile(chain.KogitoApp)
+		if err != nil {
+			chain.Error = err
+			return chain
+		}
 		chain.Resources.BuildConfigS2I = &bc
 	}
 	return chain
 }
 
 func buildConfigRuntimeBuilder(chain *builderChain) *builderChain {
-	if !chain.KogitoApp.Spec.IsGitURIEmpty() {
-		bc, err := newBuildConfigRuntime(
-			chain.KogitoApp,
-			chain.Resources.BuildConfigS2I,
-		)
-		if err != nil {
-			chain.Error = err
-			return chain
-		}
-		chain.Resources.BuildConfigRuntime = &bc
+	bc, err := newBuildConfigRuntime(
+		chain.KogitoApp,
+		chain.Resources.BuildConfigS2I,
+	)
+	if err != nil {
+		chain.Error = err
+		return chain
 	}
+
+	// if gitURI is empty, instantiate a s2i binary build
+	if chain.KogitoApp.Spec.IsGitURIEmpty() {
```

Comment:
Is this needed?
IMHO the binary build is initialized in every case in `buildConfigRuntimeBinaryBuilder` right below.

Reply:
for now I would say yes, I prefer change the binary build behaviour on the other jira that is opened to do the uploading thing on the binary builds as well.